corehq/middleware.py

```diff
@@ -210,3 +216,69 @@ def process_view(self, request, view_func, view_args, view_kwargs):
 
             if getattr(request, 'domain', None):
                 scope.set_tag('domain', request.domain)
+
+
+session_logger = logging.getLogger('session_access_log')
+
+
+def log_call(func):
+    def with_logging(*args, **kwargs):
+        session_logger.info("\n\n\n")
+        session_logger.info(func.__name__ + " was called")
+        session_logger.info("\n\n\n")
+        for line in traceback.format_stack():
+            white_list = ['corehq/', 'custom/', func.__name__]
+            if any([c in line for c in white_list]):
+                session_logger.info(line.strip())
+        return func(*args, **kwargs)
+    return with_logging
+
+
+def decorate_all_methods(decorator):
+    def decorate(cls):
+        for attr in dir(cls):
+            if "__" not in attr and callable(getattr(cls, attr)):
+                setattr(cls, attr, decorator(getattr(cls, attr)))
+        return cls
+    return decorate
+
+
+@decorate_all_methods(log_call)
+class LoggingSessionStore(SessionStore):
+    pass
+
+
+class LoggingSessionMiddleware(SessionMiddleware):
+
+    def __init__(self, get_response=None):
+        super().__init__(get_response)
+        regexes = getattr(settings, 'SESSION_BYPASS_URLS', [])
+        self.bypass_re = [
+            re.compile(regex.format(domain=legacy_domain_re)) for regex in regexes
+        ]
+
+    def _bypass_sessions(self, request, domain):
+        return (toggles.BYPASS_SESSIONS.enabled(domain) and
+            any(rx.match(request.path_info) for rx in self.bypass_re))
+
+    def process_request(self, request):
+        try:
+            match = re.search(r'/a/[0-9a-z-]+', request.path)
+            if match:
+                domain = match.group(0).split('/')[-1]
+                if self._bypass_sessions(request, domain):
+                    session_key = request.COOKIES.get(settings.SESSION_COOKIE_NAME)
+                    request.session = self.SessionStore(session_key)
```

Comment:
instead of these 2 lines you could call the super method.

Reply:
There was some edge case that prevented this, so I rolled it back to this state.